gateway/ln-gateway/src/lnd.rs

```diff
@@ -203,6 +205,30 @@ impl GatewayLndClient {
             ))
         })
     }
+
+    async fn lookup_payment(
+        &self,
+        payment_request: String,
+        _invoice_timestamp: u64,
+        client: &mut LndClient,
+    ) -> Result<Option<Payment>, GatewayError> {
+        let payments = client
+            .lightning()
+            .list_payments(ListPaymentsRequest {
+                // TODO: Returning ALL payments here is bad, because there could be a lot of
+                // payments We can't filter by timestamp until we fix our tonic_lnd
+                // dependency
+                include_incomplete: true,
+                ..Default::default()
```

Comment:
The only way I could figure out to get the status of the current payment is to list all payments. This is not good if the node has done a lot of payments in the past, since it will return all of them. I'd like to filter based on the invoice's creation timestamp, but our version of tonic_lnd is not up to date enough.

It looks like we currently maintain our own fork of `tonic_lnd`. @okjodom do you remember why we are maintaining our own fork?

Regardless, it doesn't look like the upstream repo even has the most up to date proto file https://github.com/Kixunil/tonic_lnd/blob/master/vendor/lightning.proto

It seems to me that we should work to eliminate our dependency on tonic_lnd, and especially not maintain our own fork. Updating these dependencies to take advantage of the latest LND features seems really cumbersome.

Reply:
Is the main advantage of using `tonic_lnd` that it has all of the legwork for generating the rust code from the proto files?

Reply:
In the past I've used this fork: https://github.com/yzernik/tonic_openssl_lnd/tree/master

It had all the features at the time.

It seems we're maintaining our own version because upstream is unwilling to upgrade stuff: https://github.com/Kixunil/tonic_lnd/pull/31

Reply:
@m1sterc001guy but why not use https://lightning.engineering/api-docs/api/lnd/lightning/lookup-invoice here?

Reply:
@m1sterc001guy I think the solution here is to use `.router().track_payment_v2` (https://lightning.engineering/api-docs/api/lnd/router/track-payment-v2)

Reply:
> It looks like we currently maintain our own fork of tonic_lnd. @okjodom do you remember why we are maintaining our own fork?

So we can make faster updates to the proto file if we need to. Should I bump the `lightning.proto` file to match LND version we use in dev?

Reply:
Also, I added features to `tonic_lnd` allowing us to have a slim dependency rather than all pulling in rpcs we don't need. These are available in out fork but have not landed in upstream : https://github.com/Kixunil/tonic_lnd/pull/29

Reply:
Changed to use `TrackPaymentV2` which solves the above problems